panimg/post_processors/mha_pixel_range.py

```diff
@@ -0,0 +1,39 @@
+from typing import Set
+
+import SimpleITK
+
+from panimg.models import ImageType, PanImgFile, PostProcessorResult
+
+MINIMUM_VALUE_TAG = "SmallestImagePixelValue"
+MAXIMUM_VALUE_TAG = "LargestImagePixelValue"
+
+
+def mha_pixel_range(*, image_files: Set[PanImgFile]) -> PostProcessorResult:
+    reader = SimpleITK.ImageFileReader()
+    writer = SimpleITK.ImageFileWriter()
+
+    for file in image_files:
+        if file.image_type != ImageType.MHD:
+            continue
+
+        reader.SetFileName(str(file.file.absolute()))
+
+        reader.ReadImageInformation()
+        if reader.HasMetaDataKey(MINIMUM_VALUE_TAG) and reader.HasMetaDataKey(
+            MAXIMUM_VALUE_TAG
+        ):
+            continue
+
+        image = reader.Execute()
+
+        # Use the numpy-array route to support a larger range of data-types
+        # than ITK' MinimumMaximumImageFilter (e.g. 2D uint8)
+        array = SimpleITK.GetArrayViewFromImage(image)
+        image.SetMetaData(MINIMUM_VALUE_TAG, str(array.min()))
+        image.SetMetaData(MAXIMUM_VALUE_TAG, str(array.max()))
+
+        writer.SetFileName(str(file.file.absolute()))
+        writer.Execute(image)
+
+    # No files or folders have been added
+    return PostProcessorResult(new_image_files=set(), new_folders=set())
```

Comment:
A post-processor is not the correct place for this as it is modifying the tags, not producing something new. This should be integrated into the builders.

Reply:
Granted, the only post-processing currently present only adds files. However, several builders generate sitk files and hence made me lean towards a single common process. Which would some kind of post-process after each builder generated the file/image.

Do you remember what the original rational was to make the  _dzi_ creation a post-process instead of a step in de builders? Perhaps the problematic `pyvips` requirement?

Reply:
Sure, but that is not what post processors are intended for, you would need to add something else to do that, or add it to the MHA model which is where common MHA things are done.

DZI creation takes a long time and produces another entity, so we can separate it out with async tasks, see https://github.com/DIAGNijmegen/rse-roadmap/issues/142. Post processors are aimed for that, and things like generating thumbnails.

Reply:
Ah, I see. Thanks for the clarification! I thought it more of a single-pass pipeline component for image processing and not a post-hoc processing step.